sklearn/metrics/ranking.py

```diff
@@ -712,7 +716,7 @@ def label_ranking_average_precision_score(y_true, y_score):
         L = rankdata(scores_i[relevant], 'max')
         out += (L / rank).mean()
 
-    return out / n_samples
+    return np.average(out / n_samples, weights=sample_weight)
```

Comment:
For this to work, out needs to be an array. Alternatively you can do the weighted averaging yourself by multiplying each contribution to out and dividing by the sum of weights